selfservice/strategy/link/strategy_verification_test.go

```diff
@@ -252,5 +252,77 @@ func TestVerification(t *testing.T) {
 		t.Run("type=api", func(t *testing.T) {
 			check(t, expectSuccess(t, true, values))
 		})
+
+		// HOOK TEST
+
+		t.Run("description=test verification hook error", func(t *testing.T) {
+			var check = func(t *testing.T, actual string) {
+
+				assert.EqualValues(t, verification.StrategyVerificationLinkName, gjson.Get(actual, "active").String(), "%s", actual)
+				assert.EqualValues(t, verificationEmail, gjson.Get(actual, "methods.link.config.fields.#(name==email).value").String(), "%s", actual)
+				assertx.EqualAsJSON(t, text.NewVerificationEmailSent(), json.RawMessage(gjson.Get(actual, "messages.0").Raw))
+
+				message := testhelpers.CourierExpectMessage(t, reg, verificationEmail, "Please verify your email address")
+				assert.Contains(t, message.Body, "please verify your account by clicking the following link")
+
+				verificationLink := testhelpers.CourierExpectLinkInMessage(t, message, 1)
+
+				assert.Contains(t, verificationLink, public.URL+link.RouteVerification)
+				assert.Contains(t, verificationLink, "token=")
+
+				cl := testhelpers.NewClientWithCookies(t)
+				res, err := cl.Get(verificationLink)
+				require.NoError(t, err)
+				defer res.Body.Close()
+
+				assert.Equal(t, http.StatusBadRequest, res.StatusCode)
+				// TODO: assert retry verification after webhook failure
+				// 2nd Try
```

Comment:
Why this test tries to send verification link twice and it suppose to fail on the first try? I fail to see the difference between those two calls and my local test runs seems to fail at this point. I might miss something obvious here...

Reply:
It was an own attempt to get familiar with the code, which was forgotten to be removed. Last commit (after rebase) removes it. My own fault. Sorry for stealing your time